pkg/build/api/v1/types.go

```diff
@@ -658,6 +734,27 @@ type BuildRequest struct {
 
 	// Env contains additional environment variables you want to pass into a builder container
 	Env []kapi.EnvVar `json:"env,omitempty"`
+
+	// BuildCauses can pass information from the controllers to the buildconfig
+	// generator about who/what has triggered a build through the request.
+	BuildCauses []*BuildTriggerCauses `json:"BuildCauses,omitempty"`
```

Comment:
deployments use a slice of causes because they are level driven.  using a slice here future proofs us in case we do move to level driven, so i guess it's ok, but we're never going to have more than one in the current implementation.